### ISSUE TITLE
Can't find generated files

### ISSUE THREAD

**Issue by ckpiggy** [nuxt/nuxt.js#715]
My server framework(total.js) ask me to serve static files in `'public/'`, and I store nuxt source code in `'nuxt/'`
here is my nuxt.config.js
```
module.exports = {
  head: {
    title: 'starter',
    meta: [
      { charset: 'utf-8' },
      { name: 'viewport', content: 'width=device-width, initial-scale=1' },
      { hid: 'description', content: "Nuxt.js project" }
    ],
    link: [
      { rel: 'icon', type: 'image/x-icon', href: 'favicon.ico' }
    ]
  },
  loading: { color: '#3B8070' },
  build: {
    publicPath: '/dist'
  },
  srcDir: 'nuxt/',
  rootDir: __dirname,
  generate: {
    dir: 'public/'
  }
}
```
I call nuxt.build before I use nuxt.render as my middleware
```
const Nuxt = require('nuxt')
const config = require('../nuxt.config')
const nuxt = new Nuxt(config)

nuxt.build()
  .then(function () {
    F.middleware('nuxt', nuxt.render)
  })

```
When I restart the server the log says 'Compiled successfully ' but nothing appear in my `'public/'` folder. If I run `nuxt generate`, all the generated files generated in the `'public/'` folder, but I still can not get 404 errors after I restart server, because the file names are not match.

How can I generate static files when I am building and match the file names correctly?

**Comment by ckpiggy** [comment]
I gave up the middleware, and I manual map the route to my `'public/'` folder to serve the html files. Now the browser can get the generated files, but I got other errors: 
`SyntaxError: Unexpected token }` in vendor.bundle, 
`ReferenceError: Invalid left-hand side expression in postfix operation` in nuxt.bundle
![2017-05-14 12 51 10](https://cloud.githubusercontent.com/assets/6531171/26031400/0f81172e-38a4-11e7-9a42-175da05ddf81.png)

**Comment by alexchopin** [comment]
Hi, there is a misunderstanding with nuxt.
You have 2 options with Nuxt: Server Rendered OR Static Generated.
So if you want to serve you Static Generated App, you only need to use `npm run generate`.
Otherwise, it's like you did with nuxt as middleware.

**Comment by ckpiggy** [comment]
I think there is a misunderstanding with my question. 
First, I tried Server Rendered option, and I got these 404 error. Cant find the files in dist folder.
![2017-05-15 12 43 12](https://cloud.githubusercontent.com/assets/6531171/26035945/8b40fc40-3907-11e7-9682-91c629b41470.png)

Then I modify server tried to serve static generated content, the files can be found in the right place, but the files have some error
![2017-05-14 12 51 10](https://cloud.githubusercontent.com/assets/6531171/26035987/1b8b7ca8-3908-11e7-8edc-a8f54185231b.png)

Did I miss something? Thank you in advance
